sonic-xcvrd/xcvrd/xcvrd.py

```diff
@@ -1116,14 +1167,16 @@ def task_worker(self):
                 retries = self.port_dict[lport].get('cmis_retries', 0)
                 host_lanes_mask = self.port_dict[lport].get('host_lanes_mask', 0)
                 appl = self.port_dict[lport].get('appl', 0)
-                if state != CMIS_STATE_INSERTED and (host_lanes_mask <= 0 or appl < 1):
+                if state != CMIS_STATE_INSERTED and not self.is_decomm_pending_for_lport(lport) and (host_lanes_mask <= 0 or appl < 1):
```

Comment:
@AnoopKamath if you check the error log mesage it does not convey anything for decommsion failure. so why is that check required here?

Reply:
@AnoopKamath are you able to take the feedback from Mihir over email?

Reply:
Thanks, added decomm_pending_dict